eth_account/_utils/structured_data/hashing.py

```diff
@@ -121,147 +123,118 @@ def get_depths_and_dimensions(data, depth):
 
 def get_array_dimensions(data):
     """
-    Given an array type data item, check that it is an array and return the dimensions as a tuple.
+    Given an array type data item, check that it is an array and return the dimensions
+    as a tuple, in order from inside to outside.
 
-    Ex: get_array_dimensions([[1, 2, 3], [4, 5, 6]]) returns (2, 3)
+    Ex: get_array_dimensions([[1, 2, 3], [4, 5, 6]]) returns (3, 2)
     """
     depths_and_dimensions = get_depths_and_dimensions(data, 0)
-    # re-form as a dictionary with `depth` as key, and all of the dimensions found at that depth.
+
+    # re-form as a dictionary with `depth` as key, and all of the dimensions
+    # found at that depth.
     grouped_by_depth = {
         depth: tuple(dimension for depth, dimension in group)
         for depth, group in groupby(depths_and_dimensions, itemgetter(0))
     }
 
-    # validate that there is only one dimension for any given depth.
-    invalid_depths_dimensions = tuple(
-        (depth, dimensions)
-        for depth, dimensions in grouped_by_depth.items()
-        if len(set(dimensions)) != 1
-    )
-    if invalid_depths_dimensions:
-        raise ValidationError(
-            '\n'.join(
-                [
-                    "Depth {0} of array data has more than one dimensions: {1}".
-                    format(depth, dimensions)
-                    for depth, dimensions in invalid_depths_dimensions
-                ]
-            )
-        )
-
     dimensions = tuple(
-        toolz.first(set(dimensions))
-        for depth, dimensions in sorted(grouped_by_depth.items())
+        # check that all dimensions are the same, else use "dynamic"
+        dimensions[0] if all(dim == dimensions[0] for dim in dimensions) else "dynamic"
```

Comment:
> I'm pretty sure we were grabbed smallest dimension of the array (the first dim in a sorted order) and returned that for the dimension at that depth.

Actually, this would've been true if we even accepted more than one dimension but we were validating that only one dimension was accepted. So I suppose there used to be validation there but now we properly accept varying dimensions and label that depth as "dynamic". Wasn't sure what else would be appropriate here. There may be some better messaging we can provide either now or in the future but at the moment this should open it up to accept them.